mycore-base/src/main/java/org/mycore/common/content/MCRSourceContent.java

```diff
@@ -76,6 +84,17 @@ public MCRSourceContent(Source source) {
                     }
                 }
             }
+        } else if (source instanceof JAXBSource) {
+            try {
+                JAXBSource src = (JAXBSource) source;
+                StringWriter writer = new StringWriter();
+                TransformerFactory transformerFactory = TransformerFactory.newInstance();
+                Transformer transformer = transformerFactory.newTransformer();
+                transformer.transform(src, new StreamResult(writer));
+                baseContent = new MCRJDOMContent(new SAXBuilder().build(new StringReader(writer.toString())));
+            } catch (TransformerException | JDOMException | IOException e) {
+                throw new MCRException("Error while resolving JAXBSource", e);
+            }
```

Comment:
Should be lazier, as we do not know what code runs after it (Does it need to be parsed?):

```java
TransformerFactory transformerFactory = TransformerFactory.newDefaultInstance();
try {
    Transformer transformer = transformerFactory.newTransformer();
    ByteArrayOutputStream bout = new ByteArrayOutputStream();
    transformer.transform(source, new StreamResult(bout));
    baseContent = new MCRByteContent(bout.toByteArray());
} catch (TransformerException e) {
    throw new MCRException("Error while resolving JAXBSource", e);
}

```